packages/admin/cms-admin/src/generator/future/generateForm.ts

```diff
@@ -28,13 +29,9 @@ export function generateForm(
     const numberFields = config.fields.filter((field) => field.type == "number");
     const booleanFields = config.fields.filter((field) => field.type == "boolean");
 
-    const isFieldOptional = (fieldName: string): boolean => {
-        const schemaEntity = gqlIntrospection.__schema.types.find((type) => type.kind === "OBJECT" && type.name === gqlType);
-        if (!schemaEntity) throw new Error(`didn't find entity ${gqlType} in schema types`);
-        if (schemaEntity.kind !== "OBJECT") throw new Error(`kind of ${gqlType} is not object, but should be.`); // this should not happen
-        const fieldDef = schemaEntity.fields.find((field) => field.name === fieldName);
-        if (!fieldDef) throw new Error(`didn't find field ${fieldName} of ${gqlType} in introspected gql-schema.`);
-        return fieldDef.type.kind !== "NON_NULL";
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const isFieldOptionalWrapper = (fieldConfig: FormFieldConfig<any>) => {
```

Comment:
I think so, else it would look like this. (maybe I should rename the wrapper function to isOptional(field))

```
type FormValues = ${
        numberFields.length > 0
            ? `Omit<GQL${fragmentName}Fragment, ${numberFields.map((field) => `"${String(field.name)}"`).join(" | ")}>`
            : `GQL${fragmentName}Fragment`
    } ${
        numberFields.length > 0 || Object.keys(rootBlocks).length > 0
            ? `& {
        ${numberFields
            .map((field) => `${String(field.name)}${isFieldOptional({ config: field, gqlIntrospection, gqlType }) ? `?` : ``}: string;`)
            .join("\n")}
        ${Object.keys(rootBlocks)
            .map((rootBlockKey) => `${rootBlockKey}: BlockState<typeof rootBlocks.${rootBlockKey}>;`)
            .join("\n")}
    }`
            : ""
    };
```

Reply:
then it would look like this:

```
${numberFields.map((field) => `${String(field.name)}${isOptional(field) ? `?` : ``}: string;`).join("\n")}
```

Reply:
A correct comparison would be 

```ts
${numberFields.map((field) => `${String(field.name)}${isFieldOptional({ config: field, gqlIntrospection, gqlType }) ? `?` : ``}: string;`).join("\n")}
```
versus

```ts
${numberFields.map((field) => `${String(field.name)}${isOptional(field) ? `?` : ``}: string;`).join("\n")}
```

I don't know, one argument versus three isn't really convincing to me. But if you think it's necessary, then it's okay for me

Reply:
I got some reasons for me to prefer the wrapper version
1. the two additional params are the same for every call in this scope. in my understanding this is a practical use case for curring a function.
2. readability of `isOptional(field)` is much better than for `isFieldOptional({ config: field, gqlIntrospection, gqlType })`